src/components/modals/WalletListMenuModal.js

```diff
@@ -101,27 +101,20 @@ export function WalletListMenuModal(props: Props) {
 
   const dispatch = useDispatch()
   const account = useSelector(state => state.core.account)
-  const guiWallet = useSelector(state => state.ui.wallets.byId[walletId])
+  const edgeWallet = account.currencyWallets[walletId]
 
   const theme = useTheme()
   const styles = getStyles(theme)
 
   // Look up the image and name:
-  let image: string | void
-  let walletName: string | void
-  if (guiWallet != null) {
-    walletName = guiWallet.name
-
-    const { metaTokens } = guiWallet
-    const contractAddress = metaTokens.find(token => token.currencyCode === currencyCode)?.contractAddress
-    image = getCurrencyIcon(getPluginId(guiWallet.type), contractAddress).symbolImage
-  }
+  const walletName = edgeWallet?.name ?? ''
+  const { contractAddress } = edgeWallet?.currencyInfo.metaTokens.find(token => token.currencyCode === currencyCode) ?? {}
```

Comment:
This line doesn't pass Flow. I get "Flow does not yet support method or property calls in optional chains." In any case, passing the contract address as a tokenId is not a valid assumption, and I don't want our GUI to randomly break in a few years when we add some weird coin where that's not true.